src/core/server/cron/autoArchiving.ts

```diff
@@ -35,38 +36,62 @@ export function registerAutoArchiving(
 }
 
 const archiveStories: ScheduledJobCommand<Options> = async ({
-  log,
+  log: logger,
   mongo,
   redis,
   tenantCache,
   config,
 }) => {
   const enabled = config.get("enable_auto_archiving");
   if (!enabled) {
-    log.info("cancelling auto archiving operation as it is not enabled");
+    logger.info("cancelling auto archiving operation as it is not enabled");
     return;
   }
 
   const amount = config.get("auto_archiving_batch_size");
-  const age = config.get("auto_archive_older_than");
 
   for await (const tenant of tenantCache) {
-    log = log.child({ tenantID: tenant.id }, true);
+    const log = logger.child({ tenantID: tenant.id }, true);
 
+    const redisKey = `${REDIS_ARCHIVING_QUEUE_KEY}:${tenant.id}`;
     const now = new Date();
-    const dateFilter = new Date(now.getTime() - age);
 
     let count = 0;
     do {
-      const story = await retrieveLockedStoryToBeArchived(
+      const redisRecords = await redis.spop(redisKey, 1);
```

Comment:
Rather than using a custom integration with Redis, instead you could introduce a new Bull task that will do this immediatly instead. I suggest this because the bull tasks have a full lifecycle management. In the case that there's a failure, it can try again.

With this current implementation, if there's an error during the archiving process, it will fail without ever trying to repeat it (unless it's batched again I suppose later).

You could further cap the max number of stories that are processed through use of the limter options if that appeals to you too: https://github.com/OptimalBits/bull/tree/develop/docs#rate-limiter

Reply:
Sweet! I will look into this!

Reply:
I have updated this to use `bull`, works like a charm. Thanks for the suggestion!